### PR TITLE
Immediately reconnects to infinite tracing endpoint on 200 OK status.

### PR DESCRIPTION
**Pull request by michaelgoin**
## Proposed Release Notes

* Infinite tracing gRPC connection will now immediately reconnect on 200 OK status.

  A reconnect delay will only be used when receiving an error status that is not `UNIMPLEMENTED` (which is a permanent disconnect).

## Details

Since we only need to delay on error statuses, we didn't really need the state tracking for reconnect delay and multiple types of disconnect. Much of the red is ripping that out in favor of a simple delay configuration.

`connectSpans()` does not force an async boundary now, so there's a slightly different ordering of the `emit` that impacts tests.

### CODE REVIEW

File: lib/grpc/connection.js [hunk 1]
```diff
@@ -263,19 +222,15 @@ class GrpcConnection extends EventEmitter {
...
-        // Even an "OK status" indicates the call completed successfully,
-        // which indicates the stream is over.
-        if (grpc.status[grpc.status.OK] !== grpcStatusName) {
-          this._metrics.getOrCreateMetric(
-            util.format(NAMES.INFINITE_TRACING.SPAN_RESPONSE_GRPC_STATUS, grpcStatusName)
-          ).incrementCallCount()
-        }
+        this._metrics.getOrCreateMetric(
+          util.format(NAMES.INFINITE_TRACING.SPAN_RESPONSE_GRPC_STATUS, grpcStatusName)
+        ).incrementCallCount()
```
[the remark this answers is not in the document]
Comment (michaelgoin) [reply lib/grpc/connection.js:232]: This part of the code is for errors that are not OK and has the delay reconnect. The `else if` statement above is for the OK status and does the immediate reconnect.

I considered refactoring this code more to try to make more clear (`if else if` can be hard to follow) but decided not to go too deep. Maybe slight regrets now haha.

[the remark this answers is not in the document]
Comment (astormnewrelic) [reply lib/grpc/connection.js:232]: Ah, I see.  [Bad Diffs](https://www.youtube.com/watch?v=0qIMTA4_YdU)
